generators/client/templates/angular/src/main/webapp/app/core/auth/account.service.ts.ejs

```diff
@@ -18,19 +18,114 @@
 -%>
 import { Injectable } from '@angular/core';
 import { HttpClient, HttpResponse } from '@angular/common/http';
-import { Observable } from 'rxjs';
+import { Observable, Subject } from 'rxjs';
 
 import { SERVER_API_URL } from 'app/app.constants';
+<%_ if (websocket === 'spring-websocket') { _%>
+import { <%=jhiPrefixCapitalized%>TrackerService } from '../tracker/tracker.service';
+<%_ } _%>
 
 @Injectable({providedIn: 'root'})
 export class AccountService  {
-    constructor(private http: HttpClient) { }
+    private userIdentity: any;
+    private authenticated = false;
+    private authenticationState = new Subject<any>();
 
-    get(): Observable<HttpResponse<Account>> {
+    constructor(private http: HttpClient<% if (websocket === 'spring-websocket') { %>,
+        private trackerService: <%=jhiPrefixCapitalized%>TrackerService<% } %>) { }
+
+    fetch(): Observable<HttpResponse<Account>> {
         return this.http.get<Account>(SERVER_API_URL + '<%- apiUaaPath %>api/account', {observe : 'response'});
     }
 
     save(account: any): Observable<HttpResponse<any>> {
         return this.http.post(SERVER_API_URL + '<%- apiUaaPath %>api/account', account, {observe: 'response'});
     }
+
+    authenticate(identity) {
```

Comment:
could you add tests for this methods?

Reply:
Sure, I'll write some during the weekend